scene/resources/material.h

```diff
@@ -284,24 +284,26 @@ class BaseMaterial3D : public Material {
 
 private:
 	struct MaterialKey {
-		// enums
-		uint64_t texture_filter : TEXTURE_FILTER_MAX - 1;
-		uint64_t detail_uv : DETAIL_UV_MAX - 1;
-		uint64_t transparency : TRANSPARENCY_MAX - 1;
-		uint64_t alpha_antialiasing_mode : ALPHA_ANTIALIASING_MAX - 1;
-		uint64_t shading_mode : SHADING_MODE_MAX - 1;
-		uint64_t feature_mask : FEATURE_MAX - 1;
-		uint64_t blend_mode : BLEND_MODE_MAX - 1;
-		uint64_t depth_draw_mode : DEPTH_DRAW_MAX - 1;
-		uint64_t cull_mode : CULL_MAX - 1;
+		// enum values
+		uint64_t texture_filter : get_num_bits(TEXTURE_FILTER_MAX - 1);
+		uint64_t detail_uv : get_num_bits(DETAIL_UV_MAX - 1);
+		uint64_t transparency : get_num_bits(TRANSPARENCY_MAX - 1);
+		uint64_t alpha_antialiasing_mode : get_num_bits(ALPHA_ANTIALIASING_MAX - 1);
+		uint64_t shading_mode : get_num_bits(SHADING_MODE_MAX - 1);
+		uint64_t feature_mask : get_num_bits(FEATURE_MAX - 1);
```

Comment:
this is ok as just FEATURE_MAX since its a bitmask, its how it was originally, the rest are fine

Reply:
fixed